packages/renderer/src/lib/preferences/PreferencesResourcesRendering.spec.ts

```diff
@@ -0,0 +1,105 @@
+
+import '@testing-library/jest-dom';
+import { test, expect, vi } from 'vitest';
+import { render, screen } from '@testing-library/svelte';
+import PreferencesResourcesRendering from './PreferencesResourcesRendering.svelte';
+import { providerInfos } from '../../stores/providers';
+import type { ProviderInfo } from '../../../../main/src/plugin/api/provider-info';
+
+const providerInfo: ProviderInfo = {
+  id: 'podman',
+  name: 'podman',
+  images: {
+    icon: 'img',
+  },
+  status: 'started',
+  warnings: undefined,
+  containerProviderConnectionCreation: true,
+  detectionChecks: undefined,
+  containerConnections: [
+    {
+      name: 'machine',
+      status: 'started',
+      endpoint: {
+        socketPath: 'socket',
+      },
+      lifecycleMethods: ['start', 'stop', 'delete'],
+      type: 'podman',
+    },
+  ],
+  installationSupport: undefined,
+  internalId: '0',
+  kubernetesConnections: [],
+  kubernetesProviderConnectionCreation: true,
+  links: undefined,
+};
+
+beforeEach(() => {
+  (window.events as unknown) = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    receive: vi.fn(),
+  };
+});
+
+test('Expect to see elements regarding podman provider', async () => {
+  providerInfos.set([providerInfo]);
+  render(PreferencesResourcesRendering, {});
+  const button = screen.getByRole('button', { name: 'Create new ...' });
+  expect(button).toBeInTheDocument();
+  expect(button.title).toBe('Create new podman machine');
+});
+
+test('Expect to be start, delete actions enabled and stop, restart disabled when container stopped', async () => {
+  providerInfo.containerConnections[0].status = 'stopped';
+  providerInfos.set([providerInfo]);
+  render(PreferencesResourcesRendering, {});
+  const buttons = screen.getAllByRole('button');
+  const startButton = buttons.filter(b => b.attributes.getNamedItem('name')?.value == 'Start')[0];
+  expect(startButton).toBeInTheDocument();
+  expect(!startButton.classList.contains('cursor-not-allowed'));
+  const stopButton = buttons.filter(b => b.attributes.getNamedItem('name')?.value == 'Stop')[0];
+  expect(stopButton).toBeInTheDocument();
+  expect(stopButton.classList.contains('cursor-not-allowed'));
+  const restartButton = buttons.filter(b => b.attributes.getNamedItem('name')?.value == 'Restart')[0];
+  expect(restartButton).toBeInTheDocument();
+  expect(restartButton.classList.contains('cursor-not-allowed'));
+  const deleteButton = buttons.filter(b => b.attributes.getNamedItem('name')?.value == 'Delete')[0];
+  expect(deleteButton).toBeInTheDocument();
+  expect(!deleteButton.classList.contains('cursor-not-allowed'));
+});
+
+test('Expect to be start, delete actions disabled and stop, restart enabled when container running', async () => {
+  providerInfo.containerConnections[0].status = 'started';
+  providerInfos.set([providerInfo]);
+  render(PreferencesResourcesRendering, {});
+  const buttons = screen.getAllByRole('button');
```

Comment:
I guess we should have
```
const startButton = screen.getByRole('button', { name: 'Start' });
```

Reply:
I tried but it doesn't work that way. I don't know exactly what they mean by name but it's like it checks the text within the button. In my case i have an icon. 
I cannot use a title tag as it creates a white tooltip which has a design unrelated with the rest.
I used a name attribute for all buttons.

Reply:
did you use the `aria-label` I think it's what should be used for the attribute name (so it works for a11y and testing-library)

Reply:
works fine with the `aria-label`. Thanks!!